ocaml/xapi/xapi_pci_helpers.ml

```diff
@@ -80,11 +105,7 @@ let get_host_pcis () =
                           ; name = lookup_class_name access d.device_class
                                    |> default ~msg:"class name" }
           in
-          let related_devs =
-            List.filter (fun d' ->
-                let slot x = (x.domain, x.bus, x.dev) in
-                slot d' = slot d && d' <> d
-              ) devs in
+          let related_devs = List.filter (is_related_to d) devs in
```

Comment:
I think this will now trigger _lots_ of (redundant) `readlink` calls to the file system? And similar calls are also done higher up in xapi_pci.ml. It may be better to add the `physfn` to `type pci` in a first pass over the devices.

Reply:
The `readlink` is only called when devices are likely to be related; the `pci` type being used here is not the one defined above but in `ocaml/pci` - I'm a bit reluctant to extend it because it looks like it was vendored but I'll take a look.